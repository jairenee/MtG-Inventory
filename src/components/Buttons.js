import React from 'react'
import { Button } from 'react-bootstrap'

const ipcRenderer = window.require("electron").ipcRenderer;

export class SyncButton extends React.Component {
    constructor(props) {
        super(props);

        this.getData = this.getData.bind(this);
    }

    getData() {
        ipcRenderer.send(`store-${this.props.type}`)
        ipcRenderer.once(`${this.props.type}-stored`, (event, args) => {
            console.log("Stored");
        })
    }

    render() {
        return <Button bsSize="xsmall" onClick={this.getData}>Sync {this.props.type[0].toUpperCase() + this.props.type.substring(1)}</Button>
    }
}