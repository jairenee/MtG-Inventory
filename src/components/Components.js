export {TopNav} from "./TopNav";
export {Home} from "./Home";
export {Sets} from "./Sets";
export {Cards} from "./Cards";