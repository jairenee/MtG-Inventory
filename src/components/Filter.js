import React from 'react'
import { DropdownButton, MenuItem, InputGroup, Button, FormControl, HelpBlock } from 'react-bootstrap'

export class Filter extends React.Component {
    render() {
        let button;

        if (this.props.button) {
            button = (
                <InputGroup.Button>
                    <Button type="submit">Go!</Button>
                </InputGroup.Button>
            )
        }

        return (
            <form onSubmit={this.props.onSubmit}>
                <InputGroup>
                    <InputGroup.Button>
                        <DropdownButton
                            bsStyle="default"
                            title={this.props.defaultFilter}
                            id="search-filter"
                        >
                            {
                                this.props.filters.map((filter, key) => {
                                    return <MenuItem key={key} eventKey={filter} onSelect={this.props.onSelect}>{filter}</MenuItem>
                                })
                            }
                        </DropdownButton>
                    </InputGroup.Button>
                    <FormControl 
                        type="text" 
                        placeholder="Filter" 
                        defaultValue={this.props.defaultText} 
                        onChange={this.props.onChange} 
                        inputRef={this.props.thisRef} 
                    />
                    {button}
                </InputGroup>
            </form>
        )
    }
}

export class SetList extends React.Component {
    render() {
        return (
            <InputGroup className="filter-list">
                <FormControl 
                    type="text" 
                    size="35" 
                    defaultValue={this.props.defaultText} 
                    placeholder="Set List (CSV) e.g. m19,kld,aer" 
                    onChange={this.props.onChange} 
                    inputRef={this.props.thisRef} />
                <HelpBlock>Use this to filter lists to specific sets ordered by release date. Useful for sorting large amounts of cards!</HelpBlock>
            </InputGroup>
        )
    }
}

// export class AddFilter extends React.Component {
//     render() {

//     }
// }