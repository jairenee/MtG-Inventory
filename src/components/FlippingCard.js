import React from 'react'
import Flippy, { FrontSide, BackSide } from 'react-flippy';
import { tada as animation } from 'react-animations';
import { StyleSheet, css } from 'aphrodite';

let ipcRenderer = window.require("electron").ipcRenderer;

export default class FlippingCard extends React.Component {
    constructor(props) {
        super(props);
        this.state = {saidHi: false, flipStyle: {}};
        this.state = {backSide: ""}
    }

    componentDidMount() {
        console.log(this.props)
        ipcRenderer.send("get-back-side", {set: this.props.set, number: this.props.number})
        ipcRenderer.on(`back-side-${this.props.number}`, (event, imageUrl) => {
            this.setState({backSide: imageUrl})
        })
    }

    render() {
        var VisibilitySensor = require('react-visibility-sensor');

        let that = this;

        const styles = StyleSheet.create({
            flippable: {
                animationName: animation,
                animationDuration: '1s'
            }
        })

        var onChange = function (isVisible) {
            if (isVisible && !that.state.saidHi) {
                that.setState({flipStyle: styles.flippable, saidHi: true});
            }
        };

        return (
        <VisibilitySensor delayedCall={true} onChange={onChange}>
            <Flippy style={{width: 200}}>
                <FrontSide style={{padding:0, margins:0}}>
                    <img alt={`${this.props.set}-${this.props.number.replace(/a|b/, "")} (Front Side)`} 
                         style={{borderRadius: 10}} className={css(this.state.flipStyle)} 
                         src={this.props.thisSide}></img>
                </FrontSide>
                <BackSide style={{padding:0, margins:0}}>
                    <img alt={`${this.props.set}-${this.props.number.replace(/a|b/, "")} (Back Side)`}
                         style={{borderRadius: 10}} 
                         src={this.state.backSide}></img>
                </BackSide>
            </Flippy>
        </VisibilitySensor>
        )
    }
}