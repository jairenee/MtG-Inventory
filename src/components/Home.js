import React from 'react'

export class Home extends React.Component {
    // TODO: Make this pretty
    //       Quick search
    //       Recent decks
    //       Quick add to inventory
    render() {
      return (
        <div>
            <center>
                <h2>MtG Inventory</h2>
                <p className="lead">A tool for quickly finding and sorting information about MtG Cards</p>
            </center>
        </div>
      )
    }
  }