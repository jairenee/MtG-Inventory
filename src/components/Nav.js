import React from 'react'
import { Link } from 'react-router-dom'

export class Nav extends React.Component {
    render() {
      return (
        <nav className="navbar navbar-default">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
                <span className="sr-only">Toggle navigation</span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
              </button>
              <Link className="navbar-brand" to="/">MtG Inventory</Link>
            </div>
            <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
              <ul className="nav navbar-nav">
                <li><Link to="/">Home</Link></li>
                <li><Link to="/sets">Sets</Link></li>
                <li><Link to="/cards">Cards</Link></li>
              </ul>
            </div>
          </div>
        </nav>
      )
    }
  }