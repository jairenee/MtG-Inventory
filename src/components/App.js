import React from 'react'
import { Home, Sets, Cards, TopNav } from "../containers/Components"
import { BrowserRouter, Route } from "react-router-dom"

export default class App extends React.Component {
    render() {
      return (
        <BrowserRouter>
            <div>
                <TopNav />
                <Route exact path="/" component={Home}/>
                <Route path="/sets" component={Sets}/>
                <Route path="/cards" component={Cards}/>
            </div>
        </BrowserRouter>
      )
    }
  }