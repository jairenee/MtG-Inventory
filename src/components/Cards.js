import React from 'react'
import BootstrapTable from 'react-bootstrap-table-next'
import { Filter } from './Filter'

const ipcRenderer = window.require("electron").ipcRenderer;

export default class Cards extends React.Component {
    constructor(props) {
        super(props)
        this.searchState = new React.createRef();
        this.updateSearchTerm = this.updateSearchTerm.bind(this);
        this.dropdownSelected = this.dropdownSelected.bind(this);
        this.getAndFilter = this.getAndFilter.bind(this);
    }
    
    // Grabbing the text box.
    updateSearchTerm(event) {
        this.props.dispatch({type: "updateSearch", search: event.target.value})
    }
    
    // Form input
    // TODO: Live searching, like the sets page.
    // That will need caching. Decided to use NeDB for now.
    async getAndFilter(event){
        if (event) event.preventDefault();
        this.props.dispatch({type: "setLoading", loading: true})
        let filter = this.props.filter.toLowerCase();
        ipcRenderer.send("get-cards", {search: this.props.search, filter: filter})
        ipcRenderer.once("cards-returned", (event, list) => {
            console.log("Returned")
            this.props.dispatch({type: "handleSearch", sets: list, loading: false});
        });
    }

    componentWillReceiveProps(nextProps) {
        console.log(nextProps.filter);
        if (nextProps.updateView) {
            this.getAndFilter();
            this.props.dispatch({type: "updateView", updateView: false})
        }
    }
    
    // Handling the dropdown's filter state
    dropdownSelected(eventKey, event) {
        event.preventDefault();
        this.props.dispatch({type: "setCardsFilter", filter: eventKey, updateView: true});
    }
    
    render() {
        const columns = [{
            dataField: 'set',
            text: 'Set'
        },{
            dataField: 'number',
            text: '#'
        }, {
            dataField: 'name',
            text: 'Name'
        }, {
            dataField: "cmc",
            text: "CMC"
        }, {
            dataField: 'reverse',
            text: 'Reverse'
        }, {
            dataField: 'printings',
            text: 'Printings'
        },{
            dataField: 'type',
            text: 'Type'
        },{
            dataField: 'color',
            text: 'Color'
        },{
            dataField: 'rarity',
            text: 'Rarity'
        },{
            dataField: 'image',
            text: 'Image',
            formatter(cell) {
                return <img alt="No Card Image Available" aria-hidden src={cell}></img>
            }
        }];

        let results;

        if (this.props.data && !this.props.loading) {
            results = (
                <div className="results">
                    <h2>Cards</h2>
                    <BootstrapTable keyField="id" data={this.props.data} columns={columns} bordered={false}></BootstrapTable>
                </div>
            )
        } else if (this.props.loading) {
            results = (
                <div className="row">
                    <div className="loading-img col-sm-6 col-sm-offset-3">
                        <img className="img-responsive center-block" alt="Loading" src="loading.gif"></img>
                    </div>
                </div>
            )
        }

        return (
            <div className="filter-list col-sm-10 col-sm-offset-1">
                <Filter 
                    defaultFilter={this.props.filter}
                    button filters={["Name", "Set", "CMC"]}
                    defaultText={this.props.search}
                    thisRef={this.searchState}
                    onSubmit={this.getAndFilter} 
                    onChange={this.updateSearchTerm} 
                    onSelect={this.dropdownSelected} 
                />
                {results}
            </div>
        )
    }
}