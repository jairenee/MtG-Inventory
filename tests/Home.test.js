import React from "react"
import {Home} from "../src/components/Home"
import { shallow } from "enzyme"
import renderer from 'react-test-renderer'

describe("Home component", () => {
    it("should match the snapshot", () => {
        let tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("says name of app", () => {
        const wrapper = shallow(<Home />);
        let text = wrapper.find("h2").text();
        expect(text).toEqual("MtG Inventory")
    })
})