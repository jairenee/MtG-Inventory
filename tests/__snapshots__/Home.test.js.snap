// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div
  className="filter-list"
>
  <center>
    <h2>
      MtG Inventory
    </h2>
    <p
      className="lead"
    >
      A tool for quickly finding and sorting information about MtG Cards
    </p>
  </center>
</div>
`;
